nexus/src/app/sagas/snapshot_create.rs

```diff
@@ -207,58 +188,17 @@ impl NexusSaga for SagaSnapshotCreate {
         ));
 
         // Allocate region space for snapshot to store blocks post-scrub
-        builder.append(Node::action(
-            "datasets_and_regions",
-            "RegionsAlloc",
-            REGIONS_ALLOC.as_ref(),
-        ));
-
-        builder.append(Node::action(
-            "regions_ensure",
-            "RegionsEnsure",
-            REGIONS_ENSURE.as_ref(),
-        ));
-
-        builder.append(Node::action(
-            "created_destination_volume",
-            "CreateDestinationVolumeRecord",
-            CREATE_DESTINATION_VOLUME_RECORD.as_ref(),
-        ));
-
-        // Create the Snapshot DB object
-        builder.append(Node::action(
-            "created_snapshot",
-            "CreateSnapshotRecord",
-            CREATE_SNAPSHOT_RECORD.as_ref(),
-        ));
-
-        // Send a snapshot request to a sled-agent
-        builder.append(Node::action(
-            "snapshot_request",
-            "SendSnapshotRequest",
-            SEND_SNAPSHOT_REQUEST.as_ref(),
-        ));
-
+        builder.append(regions_alloc_action());
+        builder.append(regions_ensure_action());
+        builder.append(create_destination_volume_record_action());
+        builder.append(create_snapshot_record_action());
+        builder.append(send_snapshot_request_action());
```

Comment:
Can this use the same pattern as `register_actions` above where you don't have to specify all the actions?

Reply:
Ah, good question. This is a part I didn't feel comfortable automating yet.

In the case of *this particular saga* - `snapshot_create` - we just call all the actions that we declared, in order.

However, not all our sagas are this simple - the DAG in `make_saga_dag` describes the control flow of execution, which may be more complex than "all the actions we declared in the order of declaration".

For example...

... Constants can be embedded into the DAG:

https://github.com/oxidecomputer/omicron/blob/2f453125d0ecb95f091706976368589c073a7d35/nexus/src/app/sagas/instance_create.rs#L142-L147

... Nodes that are shared among multiple sagas can be called (such as generating UUIDs):

https://github.com/oxidecomputer/omicron/blob/2f453125d0ecb95f091706976368589c073a7d35/nexus/src/app/sagas/instance_create.rs#L149-L153

... Some nodes can be called in parallel, or called as subsagas:

https://github.com/oxidecomputer/omicron/blob/2f453125d0ecb95f091706976368589c073a7d35/nexus/src/app/sagas/instance_create.rs#L223-L245